### DISCUSSION TITLE
Not receiving packets on mac

### DISCUSSION THREAD

**Discussion by hofill** [libpnet/libpnet#693] [Q&A]
I am sending a TCP SYN packet, and even though I am receiving one back (wireshark), I am not able to print it out using libpnet. The same code works on linux, though. I've tried a similar script using python's scapy and I am getting back the answers. What can I do to fix this? Below is the code that hangs on a mac (and prints something on linux):

```rust
let (mut tx, mut rx) = transport::transport_channel(4096, Layer3(Tcp))
        .expect("Error: could not create transport channel");
//~snip~ (send tcp syn packet)
// receive packet
let mut packet_iter = transport::ipv4_packet_iter(&mut rx);
loop {
    match packet_iter.next() {
        Ok((packet, _)) => {
            println!("Received packet: {:?}", packet);
            break;
        }
        Err(e) => {
            println!("Error: {:?}", e);
        }
    }
}
```

**Comment by thiezn** [comment]
I suspect you are running this on MacOS?

MacOS network stack uses the BSD stack which doesn't return tcp or udp packets on raw sockets using layer 3 or 4.

You will have to drop down to layer 2 and work directly with Ethernet frames.
